src/idl2json/src/test.rs

```diff
@@ -58,18 +59,103 @@ struct InternetIdentityInit {
     pub canister_creation_cycles_cost: Option<u64>,
 }
 
+#[derive(CandidType, Serialize, Deserialize)]
+struct ChangedInternetIdentityInit {
+    pub assigned_user_number_range: Option<(u64, u64)>,
+    // Type change
+    pub archive_module_hash: u32,
+    // Removed
+    // pub canister_creation_cycles_cost: Option<u64>,
```

Comment:
Yes; at least there might be a better way of communicating but I want to highlight that this structure does not have the `canister_creation_cycles_cost` field defined above.  That is why, when parsing an InternetIdentityInit message with **Changed**InternetIdentityInit the `canister_creation_cycles_cost` field is not recognised.  See, below,for example:

```
            "2_138_241_783":["1000000000000"], <-- not recognised
            "archive_module_hash": <-- recognised
  ["f691f269dd66aa4fc44e6916aefee03bb7feb821aef43467526974f470cd4b07"]
```

Reply:
Note however that the type of `archive_module_hash` in that example intentionally does not match the type of the actual data.  The schema says that it should be a `u32`.  The actual data is an optional binary blob.  The actual data is shown rather than coercing the binary into some kind of number.  The aim is that the proposal data should always be preserved and that it will be shown even if the schema doesn't match.

Reply:
aaaahhh, ok, that's a very nice test. My bad I didn't understand it.